src/diffusers/modeling_utils.py

```diff
@@ -585,6 +585,37 @@ def num_parameters(self, only_trainable: bool = False, exclude_embeddings: bool
         else:
             return sum(p.numel() for p in self.parameters() if p.requires_grad or not only_trainable)
 
+    def _mps_warmup_inputs(self, batch_size=None) -> Optional[Tuple]:
+        r"""
+        Temporary procedure to run a one-time forward pass on some models, when using the `mps` device.
+
+        It has been observed that the output of some models (`unet`, `vae`) is different the first time they run than the
+        rest, for the same inputs. We are investigating the root cause of the problem, but meanwhile these methods can be
+        used, if desired, to warmup those modules so their outputs are consistent.
+
+        Return inputs suitable for the forward pass of this model.
+        These will usually be a tuple of tensors that will be automatically moved to the `mps` device on warmup.
+
+        Return `None` if no warmup is required.
+        """
+        return None
+
+    def _mps_warmup(self, batch_size=None, **kwargs):
```

Comment:
@pcuenca sorry I think my previous comments weren't super clear here. 
Do you think we could maybe remove this method all together as it's now only used in the tests and I think adding functionality to the core `modeling_utils.py` function is not really worth it just to make the tests easier. Could we maybe instead just add a generic `warm_up(...)` functions just in the testing files so that it doesn't go into the `pip` package at all?

Think it would be very nice to add as little code as possible to the general `Mixin` class in the beginning to not scare readers. 

Does this make sense?

Reply:
Yes, I agree, makes sense. I inadvertently got attached to the classes knowing the size of their inputs, it's unfortunate that we can't define those extensions separately in another file. I made the changes to the testing code but it took me a bit longer than expected because tests were giving inconsistent results for some reason.